### ISSUE TITLE
Wait until client closes all connections on `BaseConnector.close()`

### ISSUE THREAD

**Issue by atemate** [aio-libs/aiohttp#3736]
## Long story short

Now, await `BaseConnector.close()` waits until all the connections are closed instead of just sending the closing signal to all the connections. The "Unclosed connection" warnings appeared because some connections required more time to be closed because some protocols (TLS) perform additional steps on connection-close and thus require longer time.

## Expected behaviour

`BaseConnector.close()` exits only when all open connections are closed.

## Actual behaviour

`BaseConnector.close()` does not wait for its all open connections to be closed.

## Steps to reproduce
TBD

## Your environment
aiohttp 3.5.4

**Comment by thinkingerrol** [comment]
Reproduced sporadically with aiohttp 3.6.2 on Ubuntu 18.04.5, Python 3.6.9.
The server was a simple HTTP server running on localhost, without TLS (see the `ConnectionKey` below)
Client code looked something like this:
```
    async def post(self, url, json):
        async with aiohttp.ClientSession() as session:
            return await session.post(url=url, json=json)

    async def put(self, url, json):
        async with aiohttp.ClientSession() as session:
            return await session.put(url=url, json=json)

    async def delete(self, url):
        async with aiohttp.ClientSession() as session:
            return await session.delete(url=url)
```

Not sure which of these leaked a connection, but I got this error via the logging.FileHandler that I added:
```
2020-10-07 12:02:34,069 ERROR /usr/lib/python3.6/asyncio/base_events.py:1285 def default_exception_handler: Unclosed connection
client_connection: Connection<ConnectionKey(host='127.0.0.1', port=8080, is_ssl=False, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=None)>
```

**Comment by dzhalaevd** [comment]
Hi, @atemate, there is no activity here for the last 2 years, let's close this issue for now and open it again if necessary

**Comment by Dreamsorcerer** [comment]
Fixed in #3733. Not backported to 3.x though for backwards compatibility.
